app/routines/transfer_omniauth_data.rb

```diff
@@ -10,7 +10,21 @@ def exec(data, user)
     # This routine is not called for identity, so error out
     raise Unexpected if data.provider == 'identity'
 
+    user.username   = data.nickname
+    user.first_name = data.first_name.present? ? data.first_name : guessed_first_name(data.name)
+    user.last_name  = data.last_name.present?  ? data.last_name  : guessed_last_name(data.name)
```

Comment:
I don't think this always be called with a user where these fields aren't set yet. (see other parts of SessionsCreate).  So I don't think these should overwrite values that are already there.  So maybe these should all be `... if not blank`